consensus/pacemaker_consensus.go

```diff
@@ -0,0 +1,87 @@
+package consensus
+
+import (
+	"fmt"
+	"log"
+
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// Implementations of the type ConsensusPacemaker interface
+// SetHeight, SeetRound, SetStep are implemented for ConsensusDebugModule
+func (m *consensusModule) ClearLeaderMessagesPool() {
+	m.clearLeader()
+	m.clearMessagesPool()
+}
+
+func (m *consensusModule) ResetForNewHeight() {
+	m.round = 0
+	m.block = nil
+	m.prepareQC = nil
+	m.lockedQC = nil
+}
+
+func (m *consensusModule) ReleaseUtilityContext() error {
+	if m.utilityContext != nil {
+		if err := m.utilityContext.Release(); err != nil {
+			log.Println("[WARN] Failed to release utility context: ", err)
+			return err
+		}
+		m.utilityContext = nil
+	}
+
+	return nil
+}
+
+func (m *consensusModule) BroadcastMessageToNodes(msg *anypb.Any) error {
+	msgCodec, err := codec.GetCodec().FromAny(msg)
+	if err != nil {
+		return err
+	}
+
+	broadcastMessage, ok := msgCodec.(*typesCons.HotstuffMessage)
+	if !ok {
+		return fmt.Errorf("failed to cast message to HotstuffMessage")
+	}
+	m.broadcastToValidators(broadcastMessage)
+
+	return nil
+}
+
+func (m *consensusModule) IsLeader() bool {
+	return m.leaderId != nil && *m.leaderId == m.nodeId
+}
+
+func (m *consensusModule) IsLeaderSet() bool {
+	return m.leaderId != nil
+}
+
+func (m *consensusModule) NewLeader(msg *anypb.Any) error {
```

Comment:
Rename to `ElectNextLeader`

Reply:
We just changed it to `NewLeader` from `ElectNextLeader` after your review, and I think changing to `NewLeader` was a good idea.  Wdyt?

Reply:
Sorry. This isn't a big deal. Let's keep as is.